staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -100,12 +132,32 @@ func (ag *Config) GetConfig() *KubeConfig {
 }
 
 func (ag *Config) GetConfigInPlace(pos int) *KubeConfig {
+	ag.mux.RLock()
+	defer ag.mux.RUnlock()
 	if pos >= 0 && len(ag.config) > pos {
 		return ag.config[pos]
 	}
 	return nil
 }
 
+// WatchUpdate returns a bcast.Member pointer that are used to watch
+// KubeConfig update
+func (ag *Config) WatchUpdate() *bcast.Member {
+	if ag.downstreamNotifyChGrp != nil {
+		return ag.downstreamNotifyChGrp.Join()
+	}
+
+	ag.mux.RLock()
+	defer ag.mux.RUnlock()
+	if ag.downstreamNotifyChGrp != nil {
```

Comment:
The statement is the same as the above. Duplicate?

Reply:
This is to ensure while waiting for mux, downstreamNotifyChGrp is not initialized by another thead.